### PR TITLE
Add Pokémon Red and Blue

### PR DESCRIPTION
**Pull request by Alchav**
Putting this here now for when/if Pokémon Red and Blue are added to AP

### CODE REVIEW

**Comment by TheT113** [comment]
I'm curious about the logic of having elite_four_condition be random-low. Async games go longer, so random-high or even like, random-range-4-8 would make the most sense to me. random-low combined with viridian_gym_condition random-high also makes viridian gym a rarity?

Similarly for Oak's Aides, they seem really low. It's an async, we have time to go catch a bunch of Pokemon!

exp_modifier: half seems cruel. I think "normal" is slow enough; if people want to challenge themselves, they would be submitting a custom.

randomize_wild_pokemon and etc, I think should have a higher chance of vanilla. There is only 1/3 a chance of vanilla, when maybe 50% would be better. similar for randomize_pokemon_stats, randomize_trainer_parties, etc.

randomize_pokemon_types randomize_type_matchup_types, randomize_type_matchup_type_effectiveness seem extremely spicy to me and also be a lower chance than vanilla.

I could also see adding a trigger that if pokemon are rolled vanilla (maybe randomize_wild_pokemon), that TMs should also be vanilla? But that's not one I'm pushy one.

**Comment by Alchav** [comment]
> I'm curious about the logic of having elite_four_condition be random-low. Async games go longer, so random-high or even like, random-range-4-8 would make the most sense to me. random-low combined with viridian_gym_condition random-high also makes viridian gym a rarity?
> 
> Similarly for Oak's Aides, they seem really low. It's an async, we have time to go catch a bunch of Pokemon!
> 
> exp_modifier: half seems cruel. I think "normal" is slow enough; if people want to challenge themselves, they would be submitting a custom.
> 
> randomize_wild_pokemon and etc, I think should have a higher chance of vanilla. There is only 1/3 a chance of vanilla, when maybe 50% would be better. similar for randomize_pokemon_stats, randomize_trainer_parties, etc.
> 
> randomize_pokemon_types randomize_type_matchup_types, randomize_type_matchup_type_effectiveness seem extremely spicy to me and also be a lower chance than vanilla.
> 
> I could also see adding a trigger that if pokemon are rolled vanilla (maybe randomize_wild_pokemon), that TMs should also be vanilla? But that's not one I'm pushy one. (EDIT: I realized I misread the trigger at the end, which is instead based on if types are randomized... I guess that does make more sense in the long run. It's a little bit weird to have, types randomized + vanilla movesets be a possibility as well, honestly...)

Elite four condition is random high, victory road condition is random low. IIRC these mirror the crystals for GT and crystals for Ganon settings for the lttp filler

**Comment by TheT113** [comment]
Ah, my bad. Misread those ones. While those are ok, range tends to be a little more preferable just to avoid the extremely low chance of rolling a 1 on elite 4 condition.

File: games/Pokemon Red and Blue.yaml [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+    0: 0 # minimum value
+    255: 0 # maximum value
+    random: 0
+    random-low: 0
+    random-high: 0
+    half: 25
+    normal: 75
+    double: 75
+    triple: 50
+    quadruple: 25
```
Comment (t3hf1gm3nt) [inline games/Pokemon Red and Blue.yaml:141]: half seems too mean, but triple and quadruple seems too much. i'd probably do either 50/50 or 75/25 for normal/double and 0 out the rest

File: games/Pokemon Red and Blue.yaml [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+      options:
+        Pokemon Red and Blue:
+          trainer_name:
+            BLUE: 5
+            GARY: 3
+            JOHN: 1
+          rival_name:
+            RED: 5
+            ASH: 3
+            JACK: 1
```
Comment (t3hf1gm3nt) [inline games/Pokemon Red and Blue.yaml:308]: is there a reason this needs to be in triggers? couldn't this just be in the base yaml as one option to save some lines?

Comment (Alchav) [reply games/Pokemon Red and Blue.yaml:309]: I don't know how else to achieve this effect

Comment (t3hf1gm3nt) [reply games/Pokemon Red and Blue.yaml:309]: OH i was misreading exactly how this was set up. i thought they were the exact same for each version, but now see that the options are flipped depending on what version is rolled. disregard

**Comment by TheT113** [comment]
viridian_gym_condition is set to random-range-high-4-8, but the maximum is 7.

**Comment by Alchav** [comment]
> viridian_gym_condition is set to random-range-high-4-8, but the maximum is 7.
> 
> How does the exp_modifier work with normal numbers? Like what is "normal" and "double" equivalent to

I knew that, I was just seeing if you knew 👀

**Comment by Alchav** [comment]
> How does the exp_modifier work with normal numbers? Like what is "normal" and "double" equivalent to

16 is normal, 32 is double

**Comment by TheT113** [comment]
Since the Oak's Aide route 15 check is missing it's logic rule, it'd probably be good to toss it in excluded locations for now, unless you're adamant a new AP bugfix release will come out before the next async.

**Comment by Alchav** [comment]
> Since the Oak's Aide route 15 check is missing it's logic rule, it'd probably be good to toss it in excluded locations for now, unless you're adamant a new AP bugfix release will come out before the next async.

It sounds like there another AP release will occur in December, so I would guess this would be fixed before the next async

**Comment by Alchav** [comment]
Converted to draft for now as, if I get my way, there will be new options by 0.3.7, which I'm guessing will be out before the next async

**Comment by Alchav** [comment]
I have updated to add the new 0.3.7 options. I've also increased the chance of Pokémon Red and Blue games to be chosen due to the high popularity of this game - https://archipelago.gg/stats - however this entire list of weights seems to be quite random and a discussion about all the games' weights should probably be had.

**Review by Joethepic — COMMENTED** [review]
I don't think require item finder should be random personally. I also find it very strange how moves can't be vanilla but trianers and wild can

**Comment by t3hf1gm3nt** [comment]
my only comment on the update is that all the game weights will be re-evaluated once all the new games are in. i know pokemon is popular and we will try to balance the weights accordingly once we do so.

**Comment by Alchav** [comment]
> I don't think to require item finder should be random personally I think it should always be true at the very least always true when hidden items are on. I also find it very strange how moves can't be vanilla but trainers and wild can.

```
  triggers:
    - option_category: Pokemon Red and Blue
      option_name: randomize_pokemon_types
      option_result: vanilla
      options:
        Pokemon Red and Blue:
          randomize_pokemon_movesets:
            vanilla: 50
            prefer_types: 50
            completely_random: 15
          tm_compatibility:
            vanilla: 50
            prefer_types: 50
            completely_random: 5
            full_compatibility: 25
          hm_compatibility:
            vanilla: 50
            prefer_types: 50
            completely_random: 5
```

**Comment by Joethepic** [comment]
Ah  makes sense must of missed that

**Pull request merged**